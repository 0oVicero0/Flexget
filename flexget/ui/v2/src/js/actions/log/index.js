import oboe from 'oboe';
import { createAction } from 'utils/actions';

const PREFIX = '@flexget/log/';
export const LOG_START = `${PREFIX}LOG_START`;
export const LOG_MESSAGE = `${PREFIX}LOG_MESSAGE`;
export const LOG_DISCONNECT = `${PREFIX}LOG_DISCONNECT`;
export const LOG_LINES = `${PREFIX}LOG_LINES`;
export const LOG_QUERY = `${PREFIX}LOG_QUERY`;
export const LOG_CLEAR = `${PREFIX}LOG_CLEAR`;

export function startLogStream() {
  return (dispatch, getState) => {
    const { lines, query } = getState().log;
    const stream = oboe({
      url: `api/server/log?lines=${lines}&search=${query}`,
      method: 'GET',
    });

    const promise = new Promise((resolve, reject) => {
      stream
        .on('start', () => dispatch(createAction(LOG_START)))
        .on('node', '{message}', message => dispatch(createAction(LOG_MESSAGE, message)))
        .done(() => resolve())
        .fail(err => reject(err));
    });

    promise.abort = () => {
      stream.abort();
      dispatch(createAction(LOG_DISCONNECT));
    };

    return promise;
  };
}

export function setLines(lines) {
  return createAction(LOG_LINES, lines);
}

export function setQuery(query) {
  return createAction(LOG_QUERY, query);
}

export function clearLogs() {
  return createAction(LOG_CLEAR);
}

