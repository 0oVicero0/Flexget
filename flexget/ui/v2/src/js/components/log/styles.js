import styled from 'styled-components';
import Paper from 'material-ui/Paper';
import theme from 'theme';

export const LogWrapper = styled(Paper)`
  padding: 2.4rem;
  display: flex;
  height: 100%;
  box-sizing: border-box;
  flex-direction: column;
  ${theme.breakpoints.up('sm')} {
    padding-top: 0;
  }
`;

export const TableWrapper = styled.div`
  width: initial;
  flex: 1;
`;
