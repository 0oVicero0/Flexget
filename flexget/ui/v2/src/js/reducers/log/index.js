import {
  LOG_START,
  LOG_MESSAGE,
  LOG_DISCONNECT,
  LOG_LINES,
  LOG_QUERY,
  LOG_CLEAR,
} from 'actions/log';

const initState = {
  messages: [],
  connected: false,
  lines: '200',
  query: '',
};

export default function reducer(state = initState, action) {
  switch (action.type) {
    case LOG_START:
      return {
        ...state,
        messages: [],
        connected: true,
      };
    case LOG_MESSAGE:
      return {
        ...state,
        messages: [action.payload, ...state.messages],
      };
    case LOG_DISCONNECT:
      return {
        ...state,
        connected: false,
      };
    case LOG_LINES:
      return {
        ...state,
        lines: action.payload,
      };
    case LOG_QUERY:
      return {
        ...state,
        query: action.payload,
      };
    case LOG_CLEAR:
      return {
        ...state,
        messages: [],
      };
    default:
      return state;
  }
}
