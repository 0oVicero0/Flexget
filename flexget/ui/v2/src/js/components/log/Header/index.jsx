import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { withStyles, createStyleSheet } from 'material-ui/styles';
import Icon from 'material-ui/Icon';
import Menu, { MenuItem } from 'material-ui/Menu';
import TextField from 'material-ui/TextField';
import Typography from 'material-ui/Typography';

const styleSheet = createStyleSheet('LogHeader', theme => ({
  root: {
    display: 'block',
    [theme.breakpoints.up('sm')]: {
      display: 'flex',
      alignItems: 'baseline',
    },
  },
  grey: {
    color: theme.palette.grey[600],
  },
  spacer: {
    flex: {
      grow: 1,
    },
  },
  textFieldWrapper: {
    display: 'flex',
    alignItems: 'baseline',
  },
  input: {
    marginLeft: 10,
    marginRight: 10,
  },
  menuIcon: {
    paddingRight: 30,
  },
  pointer: {
    cursor: 'pointer',
  },
}));

const ENTER_KEY = 13;

class Header extends Component {
  static propTypes = {
    start: PropTypes.func.isRequired,
    connected: PropTypes.bool.isRequired,
    classes: PropTypes.object.isRequired,
    setQuery: PropTypes.func.isRequired,
    query: PropTypes.string.isRequired,
    clearLogs: PropTypes.func.isRequired,
    lines: PropTypes.string.isRequired,
    setLines: PropTypes.func.isRequired,
  };

  state = {
    open: false,
  };


  componentDidMount() {
    this.start();
  }

  componentWillUnmount() {
    this.stop();
  }

  start = () => { this.stream = this.props.start(); }
  stop = () => this.stream && this.stream.abort()
  reload = () => {
    this.stop();
    this.start();
  }
  clearLogs = () => this.props.clearLogs()
  handleLines = event => this.props.setLines(event.target.value)
  handleQuery = event => this.props.setQuery(event.target.value)
  handleKeyPress = event => event.which === ENTER_KEY && this.reload()
  handleRequestClose = () => this.setState({
    open: false,
    anchorEl: undefined,
  })

  handleMenuClick = event => this.setState({
    open: true,
    anchorEl: event.currentTarget,
  })

  render() {
    const { connected, classes, query, lines } = this.props;
    const { anchorEl, open } = this.state;
    const helperText = 'Supports operators and, or, (), and "str"';

    return (
      <div className={classes.root}>
        <div>
          <Typography type="title">
            Server Log
          </Typography>
          <Typography className={classes.grey} type="subheading">
            { connected ? 'Streaming' : 'Disconnected' }
          </Typography>
        </div>
        <div className={classes.spacer} />
        <div className={classes.textFieldWrapper}>
          <Icon className={`fa fa-filter ${classes.grey}`} />
          <TextField
            id="filter"
            label="Filter"
            className={classes.input}
            value={query}
            onChange={this.handleQuery}
            inputProps={{
              onKeyPress: this.handleKeyPress,
            }}
            helperText={helperText}
          />
          <Icon onClick={this.handleMenuClick} className={`fa fa-ellipsis-v ${classes.grey} ${classes.pointer}`} />
        </div>
        <Menu
          id="log-menu"
          anchorEl={anchorEl}
          open={open}
          onRequestClose={this.handleRequestClose}
        >
          <MenuItem>
            <TextField
              id="lines"
              label="Max Lines"
              value={lines}
              onChange={this.handleLines}
              type="number"
              inputProps={{
                onKeyPress: this.handleKeyPress,
              }}
            />
          </MenuItem>
          <MenuItem onClick={this.clearLogs}>
            <Icon className={`${classes.menuIcon} fa fa-eraser`} />
            Clear
          </MenuItem>
          <MenuItem onClick={connected ? this.stop : this.start}>
            <Icon className={
              classNames(classes.menuIcon, 'fa', {
                'fa-play': !connected,
                'fa-stop': connected,
              })}
            />
            {connected ? 'Stop' : 'Start'}
          </MenuItem>
        </Menu>
      </div>
    );
  }
}

export default withStyles(styleSheet)(Header);

