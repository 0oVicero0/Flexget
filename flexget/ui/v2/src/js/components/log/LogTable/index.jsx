import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from 'material-ui/styles';
import { AutoSizer, Table, Column } from 'react-virtualized';
import 'react-virtualized/styles.css';

const styleSheet = () => ({
  table: {
    fontSize: 10,
  },
  error: {
    backgroundColor: '#f2dede',
  },
  critical: {
    backgroundColor: '#f2dede',
  },
  warning: {
    backgroundColor: '#fcf8e3',
  },
});

const LogTable = ({ messages, classes }) => (
  <AutoSizer>
    {({ height, width }) => (
      <Table
        className={classes.table}
        rowCount={messages.length}
        rowHeight={20}
        headerHeight={50}
        width={width}
        height={height}
        rowGetter={({ index }) => messages[index]}
        rowClassName={({ index }) => messages[index] && classes[messages[index]
          .log_level.toLowerCase()]}
      >
        <Column
          label="Time"
          dataKey="timestamp"
          width={100}
        />
        <Column
          label="Level"
          dataKey="log_level"
          width={100}
        />
        <Column
          label="Plugin"
          dataKey="plugin"
          width={100}
        />
        <Column
          label="Task"
          dataKey="task"
          width={100}
        />
        <Column
          label="Message"
          dataKey="message"
          width={100}
          flexGrow={1}
        />
      </Table>
    )}
  </AutoSizer>
);

LogTable.propTypes = {
  messages: PropTypes.arrayOf(PropTypes.shape({
    timestamp: PropTypes.string,
    message: PropTypes.string,
    task: PropTypes.string,
    log_level: PropTypes.string,
    plugin: PropTypes.string,
  })).isRequired,
  classes: PropTypes.object.isRequired,
};

export default withStyles(styleSheet)(LogTable);
