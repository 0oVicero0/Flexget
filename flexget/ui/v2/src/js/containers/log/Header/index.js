import { connect } from 'react-redux';
import {
  LOG_CONNECT,
  LOG_DISCONNECT,
  LOG_CLEAR,
} from 'actions/log';
import { action, request } from 'utils/actions';
import Header from 'components/log/Header';

export function mapStateToProps({ log }) {
  return {
    connected: log.connected,
  };
}

function mapDispatchToProps(dispatch) {
  return {
    start: payload => dispatch(request(LOG_CONNECT, payload)),
    stop: () => dispatch(request(LOG_DISCONNECT)),
    clearLogs: () => dispatch(action(LOG_CLEAR)),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(Header);
