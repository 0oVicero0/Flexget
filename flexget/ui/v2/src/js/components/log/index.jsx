import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from 'material-ui/styles';
import Paper from 'material-ui/Paper';
import Header from 'containers/log/Header';
import LogTable from 'containers/log/LogTable';

const styleSheet = theme => ({
  root: {
    padding: 24,
    display: 'flex',
    height: '100%',
    boxSizing: 'border-box',
    flexDirection: 'column',
    [theme.breakpoints.up('sm')]: {
      paddingTop: 0,
    },
  },
  logTable: {
    width: 'initial',
    flex: {
      grow: 1,
    },
  },
});

const LogPage = ({ classes }) => (
  <Paper className={classes.root} elevation={4}>
    <Header />
    <div className={classes.logTable}>
      <LogTable />
    </div>
  </Paper>
);

LogPage.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styleSheet)(LogPage);
