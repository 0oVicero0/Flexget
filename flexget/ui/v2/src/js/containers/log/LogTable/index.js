import { connect } from 'react-redux';
import LogTable from 'components/log/LogTable';

export function mapStateToProps({ log }) {
  return {
    messages: log.messages,
  };
}

export default connect(mapStateToProps)(LogTable);
