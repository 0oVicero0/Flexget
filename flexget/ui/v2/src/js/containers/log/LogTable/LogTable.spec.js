import { mapStateToProps } from 'containers/log/LogTable';

describe('containers/log/LogTable', () => {
  it('should return the right stuff', () => {
    expect(mapStateToProps({ log: {
      messages: [],
    } })).toMatchSnapshot();
  });
});
