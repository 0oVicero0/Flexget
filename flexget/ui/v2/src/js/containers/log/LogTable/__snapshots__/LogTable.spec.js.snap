// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/log/LogTable should return the right stuff 1`] = `
Object {
  "messages": Array [],
}
`;
